dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/ArtifactProblem.java

```diff
@@ -35,6 +35,8 @@
 
   protected final ImmutableList<DependencyNode> dependencyPath;
 
+  // todo since the artifact is the last element in the list, there's really no
+  // reason to pass the artifact here. 
   protected ArtifactProblem(Artifact artifact, List<DependencyNode> dependencyPath) {
```

Comment:
This non-public constructor allows artifact problems in which we don’t know dependencyPath.